### PR TITLE
More bounds checking in Adjust::adjustDateTime

### PR DESCRIPTION
**Pull request by kevinbackhouse**
I moved #2231 to draft because it had some build failures on Windows, caused by the code in `Adjust::adjustDateTime`. The problem is that `struct tm` has fields of type `int`, which means that there's quite a lot of implicit casting happening here.

In this PR, I've added some bounds checking, `static_cast`, and `Safe::add`, to make the conversions explicit. After this PR is merged, I'll be able to fix the build failures in #2231 more easily.

### CODE REVIEW

File: app/actions.cpp [hunk 1]
```diff
@@ -1407,19 +1409,44 @@ int Adjust::adjustDateTime(Exiv2::ExifData& exifData, const std::string& key, co
...
+
+  // bounds checking for monthAdjustment_
+  enforce<std::overflow_error>(monthAdjustment_ >= std::numeric_limits<decltype(tm.tm_mon)>::min(),
+                               "month adjustment too low");
+  enforce<std::overflow_error>(monthAdjustment_ <= std::numeric_limits<decltype(tm.tm_mon)>::max(),
+                               "month adjustment too high");
+  const auto monthAdjustment = static_cast<decltype(tm.tm_mon)>(monthAdjustment_);
+
+  // bounds checking for dayAdjustment_
+  enforce<std::overflow_error>(dayAdjustment_ >= std::numeric_limits<time_t>::min() / 86400, "day adjustment too low");
```
Comment (piponazo) [inline app/actions.cpp:1428]: [suggestion] Replace the 3 occurrences of `86400` by a constant.

**Review by piponazo — DISMISSED** [review]
Thanks for taking care of this.

At first sight all the additions look good, but I am always a bit skeptical with changes not providing its own tests to validate the implementation. Since these changes are in the application level, you cannot add simple tests in the `unitTests` target. However, I think it should be possible to add some integration tests in the python suite to exercise this new code.

I let you decide if you want to address such task in this PR or create a new issue describing the lack of tests for this new code, so that somebody else (or you in the future) want to take care of that.

**Comment by kevinbackhouse** [comment]
@piponazo: thanks for the review! I added a new test, which triggers 6 of the 8 new error messages. I don't believe it's currently possible to trigger the "seconds" error message because I can't find a command-line argument that will let me enter a large number for the seconds adjustment.

**Comment by kevinbackhouse** [comment]
The test was failing on Windows (due to the different size of the `long` type), so I've removed it from this PR and transferred it to #2231. The test passes on all platforms after the switch to `int64_t` in #2231.

**Review by piponazo — APPROVED** [review]

**Comment by piponazo** [comment]
Great! Thanks for the extra effort 💪

**Pull request merged**